src/librustc_typeck/check/mod.rs

```diff
@@ -1350,19 +1348,38 @@ pub fn check_item_type<'a,'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, it: &'tcx hir::Item
     }
 }
 
-fn check_const_is_u8_array<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
-                                     def_id: DefId,
-                                     span: Span) {
-    match tcx.type_of(def_id).sty {
-        ty::TyArray(t, _) => {
-            match t.sty {
-                ty::TyUint(ast::UintTy::U8) => return,
-                _ => {}
-            }
+fn maybe_check_static_with_link_section(tcx: TyCtxt, id: DefId, span: Span) {
```

Comment:
cc @oli-obk Should this be in `rustc_typeck`?

Reply:
Seems fine for now. I can merge it into https://github.com/rust-lang/rust/pull/51361 once that becomes a thing (since this is more of a constant value validation than a type check)